### PR TITLE
Added the Knowi Source

### PR DESCRIPTION
**Pull request by mohamad-balouza**
# Description
I created the Knowi Source which has the following streams:
- Dashboards
- Widgets
No new dependencies required.

# How Has This Been Tested?
I followed this [documentation](https://github.com/mage-ai/mage-ai/tree/master/mage_integrations) to test my source.

Note: Please let me know if you need anything else for the PR.

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Thank you for adding the new sources! Will review the PRs.

**Comment by mohamad-balouza** [comment]
> Thank you for adding the new sources! Will review the PRs.

File: mage_integrations/mage_integrations/sources/knowi/__init__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def get_forced_replication_method(self, stream_id):
+        return STREAMS[stream_id].replication_method
+
+    def get_table_key_properties(self, stream_id):
+        return STREAMS[stream_id].key_properties
+
+    def get_valid_replication_keys(self, stream_id):
+        return STREAMS[stream_id].valid_replication_keys
+
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/knowi/__init__.py:55]: Same here. Could you add the test_connection method?

File: mage_integrations/mage_integrations/sources/knowi/__init__.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        stream,
+        bookmarks: Dict = None,
+        query: Dict = None,
+        **kwargs,
+    ) -> Generator[List[Dict], None, None]:
+        if query is None:
+            query = {}
+
+        access_token = self.config.get("access_token")
+        client = KnowiClient(
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/knowi/__init__.py:24]: Also pass the logger to the client.

**Comment by mohamad-balouza** [comment]
These are the screenshots for testing the Knowi Source
Knowi Logs
Knowi test connection
Knowi UI dashboards sample data

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**